tests/saw/spec/handshake/s2n_handshake_io.cry

```diff
@@ -414,18 +429,23 @@ handshakes_fn handshk =
 // (i.e. the fields are indexing into a valid state of the handshake
 // state machine)
 valid_handshake : handshake -> Bit
-valid_handshake hs = handshakefn != zero /\
-                     (hs.message_number < length handshakefn - 1) /\
-                     ((hs.message_number > 0) ==> (handshakefn@(hs.message_number + 1) != 0)) where
-                     handshakefn = (handshakes_fn hs.handshake_type)
+
+//we use a where clause, so handshakefn is defined in the last line, go there first
+valid_handshake hs = handshakefn != zero /\                             //We must have actually found a handshake
+                     (hs.message_number < length handshakefn - 1) /\    //Our message number is not out of bounds for the handshake
+                     ((hs.message_number > 0) ==> (handshakefn@(hs.message_number + 1) != 0)) //If the message is greater than zero
+                     // We can advance the handshake and still find a next state
+                     where
+                        handshakefn = (handshakes_fn hs.handshake_type) //handshakefn is the handshake for a specific message type
```

Comment:
Instead of comments, you could move all these checks to named functions to make this clearer.

Nit: We generally align "where" one tab over in the parent function (see other definitions).

Reply:
I don't agree with the named functions here. I find the comments easier to read. Of course, I'm not sure what the named functions would look like.

By "one tab over" do you mean put the `where` in the original location? Or and additional indent on the next line?

Reply:
You probably also don't need the comment explaining where clauses. They're used basically everywhere in cryptol.

By "one tab over" I mean:
```
valid_handshake hs = handshakefn != zero /\ 
                     (hs.message_number < length handshakefn - 1) /\ 
                     ((hs.message_number > 0) ==> (handshakefn@(hs.message_number + 1) != 0))
    where handshakefn = (handshakes_fn hs.handshake_type)
```

Named functions would look like:
```
valid_handshake hs = handshake_type_valid /\ msg_num_inside_array_bounds /\ nxt_msg_exists
    where handshakefn = (handshakes_fn hs.handshake_type)
          handshake_type_valid = handshakefn != zero
          msg_num_inside_array_bounds = (hs.message_number < length handshakefn - 1)
          nxt_msg_exists = ((hs.message_number > 0) ==> (handshakefn@(hs.message_number + 1) != 0)
```